### PR TITLE
[Draft PR] Adds Stream::gt

### PR DESCRIPTION
**Pull request by assemblaj**
Blocked by #268
For issue #129
Based on : https://github.com/rust-lang/rust/blob/master/src/libcore/iter/traits/iterator.rs#L2828

### CODE REVIEW

File: src/stream/stream/mod.rs [hunk 1]
```diff
@@ -1147,6 +1149,44 @@ extension_trait! {
...
+            let single = VecDeque::from(vec![1]);
+            let single_gt = VecDeque::from(vec![10]);
+            let multi = VecDeque::from(vec![1,2]);
+            let multi_gt = VecDeque::from(vec![1,5]);
+
+            assert_eq!(single.clone().gt(single.clone()).await, false);
+            assert_eq!(single_gt.clone().gt(single.clone()).await, true);
+            assert_eq!(multi.clone().gt(single_gt.clone()).await, false);
+            assert_eq!(multi_gt.clone().gt(multi.clone()).await, true);
+
```
Comment (yoshuawuyts) [inline src/stream/stream/mod.rs]:
```suggestion
```

File: src/stream/stream/mod.rs [hunk 2]
```diff
@@ -1147,6 +1149,44 @@ extension_trait! {
         {
             FromStream::from_stream(self)
         }
+            
+        #[doc = r#"
+            Determines if the elements of this `Stream` are lexicographically
+            greater than those of another.
+            
+            # Examples
```
Comment (yoshuawuyts) [inline src/stream/stream/mod.rs]:
```suggestion
            # Examples

```

**Comment by stjepang** [comment]
Just a friendly ping: this PR needs to be rebased onto master.

**Comment by assemblaj** [comment]
@stjepang @yoshuawuyts  Rebased #304, #285 and #273. Creating new branches for #305 and #306 due to rebasing issues.

**Pull request merged**